File: backend/app/schemas/watchlist.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class WatchlistCreate(BaseModel):
    """
    Request body for creating a watchlist.
    """

    name: str


class WatchlistUpdate(BaseModel):
    """
    Request body for renaming a watchlist.
    """

    name: str


class WatchlistResponse(BaseModel):
    """
    Watchlist response.
    """

    model_config = ConfigDict(
        from_attributes=True,
    )

    id: int
    name: str
    created_at: datetime
    updated_at: datetime


class WatchlistItemCreate(BaseModel):
    """
    Request body for adding a symbol.
    """

    symbol: str


class WatchlistItemResponse(BaseModel):
    """
    Watchlist item response.
    """

    model_config = ConfigDict(
        from_attributes=True,
    )

    id: int
    symbol: str
    created_at: datetime

File: frontend/src/types/watchlist.ts
export interface Watchlist {
  id: number;
  name: string;
  created_at: string;
  updated_at: string;
}

export interface WatchlistItem {
  id: number;              // watchlist item id
  instrument_id: number;   // instrument id
  symbol: string;
  created_at: string;
}

export interface CreateWatchlistRequest {
  name: string;
}

export interface RenameWatchlistRequest {
  name: string;
}

export interface AddWatchlistItemRequest {
  symbol: string;
}